apps/indexer/lib/indexer/fetcher/celo_election_rewards.ex

```diff
@@ -0,0 +1,235 @@
+    do: after_rewards_votes - before_rewards_votes + plus_revoked_minus_activated_votes
+
+  def fetch_from_blockchain(entry) do
+    case AccountReader.active_votes(entry) do
+      {:ok, data} ->
+        data
+
+      error ->
+        Logger.debug(inspect(error))
+        Map.put(entry, :error, error)
+    end
+  end
+
+  def get_validator_and_group_rewards(
+        %{group_rewards: _group_rewards, validator_rewards: _validator_rewards} = block_with_rewards
+      ),
+      do: block_with_rewards
+
+  def get_validator_and_group_rewards(
+        %{block_number: block_number, block_timestamp: block_timestamp} = block_with_rewards
+      ) do
+    validator_and_group_rewards =
+      ValidatorEpochPaymentDistributedEvent.get_validator_and_group_rewards_for_block(block_number)
+
+    validator_rewards =
+      Enum.map(validator_and_group_rewards, fn reward ->
+        %{
+          account_hash: reward.validator,
+          amount: reward.validator_payment,
+          associated_account_hash: reward.group,
+          block_number: block_number,
+          block_timestamp: block_timestamp,
+          reward_type: "validator"
+        }
+      end)
+
+    group_rewards =
+      Enum.map(validator_and_group_rewards, fn reward ->
+        %{
+          account_hash: reward.group,
+          amount: reward.group_payment,
+          associated_account_hash: reward.validator,
+          block_number: block_number,
+          block_timestamp: block_timestamp,
+          reward_type: "group"
+        }
+      end)
+
+    Map.merge(block_with_rewards, %{validator_rewards: validator_rewards, group_rewards: group_rewards})
+  end
+
+  def import_items(block_with_rewards) do
+    reward_types_present_for_block =
+      MapSet.intersection(
+        MapSet.new(Map.keys(block_with_rewards)),
+        MapSet.new([:voter_rewards, :validator_rewards, :group_rewards])
+      )
+
+    block_with_changes =
+      reward_types_present_for_block
+      |> Enum.reduce(block_with_rewards, fn type, block_with_rewards ->
+        case changeset(Map.get(block_with_rewards, type)) do
+          {:ok, changes} -> Map.put(block_with_rewards, type, changes)
+          {:error} -> Map.drop(block_with_rewards, [type])
+        end
+      end)
+
+    case chain_import(block_with_changes) do
+      :ok -> :ok
+      {:error, :changeset} -> block_with_changes
+      {:error, :import} -> block_with_rewards
+    end
+  end
+
+  def changeset(election_rewards) do
+    {changes, all_valid} =
+      Enum.map_reduce(election_rewards, true, fn reward, all_valid ->
+        changeset = CeloElectionRewardsChain.changeset(%CeloElectionRewardsChain{}, reward)
+        log_changeset_error(changeset)
+        {changeset.changes, all_valid and changeset.valid?}
+      end)
+
+    if all_valid do
+      {:ok, changes}
+    else
+      {:error}
+    end
+  end
+
+  defp log_changeset_error(changeset) when not changeset.valid? do
```

Comment:
we can move this into the if statement above and just log before returning the error tuple